### ISSUE TITLE
Randomly getting "Exception in thread "NonBlockingInputStreamThread"" in Ruby shells

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1478]
I don't know how to reproduce this. But I'm getting the following issue very uncommonly and randomly while `irb` or `pry` is running. I use iTerm2 on macOS with bash/fish (it happened in both shells)

Error:

```
[1] pry(main)> Exception in thread "NonBlockingInputStreamThread" java.lang.AssertionError: No current context available.
	at java.lang.Throwable.<init>(Throwable.java:265)
	at java.lang.Error.<init>(Error.java:70)
	at java.lang.AssertionError.<init>(AssertionError.java:58)
	at java.lang.AssertionError.<init>(AssertionError.java:74)
	at com.oracle.truffle.polyglot.PolyglotContextImpl.requireContext(PolyglotContextImpl.java:290)
	at com.oracle.truffle.polyglot.PolyglotImpl$EngineImpl.getCurrentContext(PolyglotImpl.java:512)
	at com.oracle.truffle.api.TruffleLanguage.getCurrentContext(TruffleLanguage.java:1291)
	at org.truffleruby.RubyLanguage.getCurrentContext(RubyLanguage.java:109)
	at org.truffleruby.language.RubyBaseNode.getContext(RubyBaseNode.java:90)
	at org.truffleruby.language.methods.ExceptionTranslatingNode.translate(ExceptionTranslatingNode.java:90)
	at org.truffleruby.language.methods.ExceptionTranslatingNode.execute(ExceptionTranslatingNode.java:54)
	at org.truffleruby.language.LazyRubyNode.execute(LazyRubyNode.java:42)
	at org.truffleruby.language.RubyRootNode.execute(RubyRootNode.java:54)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callProxy(OptimizedCallTarget.java:289)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callRoot(OptimizedCallTarget.java:278)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.callBoundary(OptimizedCallTarget.java:265)
	at com.oracle.svm.truffle.api.SubstrateOptimizedCallTarget.doInvoke(SubstrateOptimizedCallTarget.java:118)
	at org.graalvm.compiler.truffle.runtime.OptimizedCallTarget.call(OptimizedCallTarget.java:223)
	at org.truffleruby.RubyContext.send(RubyContext.java:453)
	at org.truffleruby.core.adapters.InputStreamAdapter.read(InputStreamAdapter.java:29)
	at jline.internal.NonBlockingInputStream.run(NonBlockingInputStream.java:291)
	at java.lang.Thread.run(Thread.java:748)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:479)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:199)
Caused by: Attached Guest Language Frames (1)
```

**Comment by deepj** [comment]
After `ctrl` + `c` I got this

https://gist.github.com/deepj/1ba1016f8680f5aaadb1dafe07aa1ee1

Then I had to terminate it via `kill` command

**Comment by chrisseaton** [comment]
I can imagine how that's happening - should be easy enough to fix.

**Comment by eregon** [comment]
Looks due to Jline. Maybe disabling the `NonBlockingInputStream` would fix this and avoid the extra thread, since we make reading interruptible anyway IIRC (with SIGVTALRM, like MRI).

**Comment by MaxLap** [comment]
I made an issue that about a similar error, but i can reproduce it 100% of the time. #1486

**Comment by eregon** [comment]
Interestingly, I can only reproduce #1486 on SubstrateVM, not on JVM.

**Comment by eregon** [comment]
This should be fixed by 2470d862e2341aabd235e74f5d8924e1615e6d69. It should be in 1.0.0-rc14.
@deepj @MaxLap  Thanks for the report and easy reproduction steps!
